zerver/webhooks/bitbucket3/tests.py

```diff
@@ -95,16 +97,203 @@ def test_push_update_multiple_branches(self) -> None:
         expected_topic = self.EXPECTED_TOPIC_BRANCH_EVENTS.format(branch="master")
         self.send_and_test_stream_message("repo_push_update_multiple_branches",
                                           expected_topic,
                                           expected_message)
 
         self.url = self.build_webhook_url(branches='branch1')
         expected_message = """hypro999 pushed to branch branch1. Head is now \
-3980c2be32a7e23c795741d5dc1a2eecb9b85d6d"""
+3980c2be32a7e23c795741d5dc1a2eecb9b85d6d."""
         expected_topic = self.EXPECTED_TOPIC_BRANCH_EVENTS.format(branch="branch1")
         self.send_and_test_stream_message("repo_push_update_multiple_branches",
                                           expected_topic,
                                           expected_message)
+
+    # Core PR Events:
+    def test_pr_opened_without_reviewers(self) -> None:
+        expected_topic = "sandbox / PR #1 Branch1"
+        expected_message = """hypro999 opened [PR #1](http://139.59.64.214:7990/projects/SBOX/\
+repos/sandbox/pull-requests/1)\nfrom `branch1` to `master`\n\n~~~ quote\n* Add file2.txt\r\n* \
+Add file3.txt\n~~~"""
+        self.send_and_test_stream_message("pull_request_opened_without_reviewers",
+                                          expected_topic,
+                                          expected_message)
+
+    def test_pr_opened_without_description(self) -> None:
+        expected_topic = "sandbox / PR #2 Add notes feature."
+        expected_message = """hypro999 opened [PR #2](http://139.59.64.214:7990/projects/SBOX/\
+repos/sandbox/pull-requests/2)\nfrom `master` to `master`"""
+        self.send_and_test_stream_message("pull_request_opened_without_description",
+                                          expected_topic,
+                                          expected_message)
+
+    def test_pr_opened_with_two_reviewers(self) -> None:
+        expected_topic = "sandbox / PR #5 Add Notes Feature"
+        expected_message = """hypro999 opened [PR #5](http://139.59.64.214:7990/projects/SBOX/\
+repos/sandbox/pull-requests/5)(assigned to [shimura](http://139.59.64.214:7990/users/shimura) \
+and [sougo](http://139.59.64.214:7990/users/sougo))\nfrom `master` to `master`"""
+        self.send_and_test_stream_message("pull_request_opened_with_two_reviewers",
+                                          expected_topic,
+                                          expected_message)
+
+    def test_pr_opened_with_mulitple_reviewers(self) -> None:
+        expected_topic = "sandbox / PR #6 sample_file: Add sample_file.txt."
+        expected_message = """hypro999 opened [PR #6](http://139.59.64.214:7990/projects/SBOX/\
+repos/sandbox/pull-requests/6)(assigned to [sougo](http://139.59.64.214:7990/users/sougo), [zura]\
+(http://139.59.64.214:7990/users/zura) and [shimura](http://139.59.64.214:7990/users/shimura))\n\
+from `master` to `master`\n\n~~~ quote\nAdd a simple text file for further testing purposes.\
+\n~~~"""
```

Comment:
I think for PRs with reviewers, we should add the `hypro999 opened PR #6 from master to master (assigned to user1 and user2 for review)`. The reviewer information should come at the end, and we should mention that the PR has been assigned *for review* since something being assigned can mean multiple things in BitBucket. :)

Reply:
Great idea! I was originally refraining from doing something like this because of the discussion we (including Rishi Gupta and Tim Abbott) had earlier on zulipchat where I was **strongly** encouraged to strictly stick to the already existing git templates. I made the change (and honestly was quite satisfied in doing so :smiley:)